grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/services/impl/DefaultCleanJobRepository.groovy

```diff
@@ -0,0 +1,84 @@
+         // One for JcrNodeResource and one for SocialResourceWrapper
+
+         JcrJobRepositoryFactoryBean jobRepositoryFactoryBean = configurableApplicationContext.getBean(JcrJobRepositoryFactoryBean)
+
+         List<String> jobExecutionPaths = jobRepositoryFactoryBean.jobExecutionDao.getJobExecutions([BatchStatus.FAILED, BatchStatus.COMPLETED])
+         List<String> olderThanHoursJobExecutions = jobRepositoryFactoryBean.jobExecutionDao.getOlderJobExecutions(hours, jobExecutionPaths)
+         List<String> jobInstancesToRemove = jobRepositoryFactoryBean.jobInstanceDao.getJobInstancePaths(olderThanHoursJobExecutions)
+         List<String> stepExecutionsToRemove = jobRepositoryFactoryBean.stepExecutionDao.getStepExecutionPaths(olderThanHoursJobExecutions)
+         List<String> jobExecutionContextsToRemove = jobRepositoryFactoryBean.executionContextDao.getJobExecutionContextPaths(olderThanHoursJobExecutions)
+         List<String> stepExecutionContextsToRemove = jobRepositoryFactoryBean.executionContextDao.getStepExecutionContextPaths(stepExecutionsToRemove)
+
+         JcrUtil.withResourceResolver(resourceResolverFactory, "admin") { ResourceResolver resolver ->
+
+             log.info "jobInstancesToRemove: $jobInstancesToRemove, size: ${jobInstancesToRemove.size()}"
+             log.info "stepExecutionsToRemove: $stepExecutionsToRemove, size: ${stepExecutionsToRemove.size()}"
+             log.info "jobExecutionContextsToRemove: $jobExecutionContextsToRemove, size: ${jobExecutionContextsToRemove.size()}"
+             log.info "stepExecutionContextsToResource: $stepExecutionContextsToRemove, size: ${stepExecutionContextsToRemove.size()}"
+
+             log.info "Removing ${jobInstancesToRemove.size()} JobInstances"
+             removeResources(jobInstancesToRemove, resolver)
+             log.info "Removing ${olderThanHoursJobExecutions.size()} JobExecutions"
+             removeResources(olderThanHoursJobExecutions, resolver)
+             log.info "Removing ${stepExecutionsToRemove.size()} StepExecutions"
+             removeResources(stepExecutionsToRemove, resolver)
+             log.info "Removing ${jobExecutionContextsToRemove.size()} JobExecutionContexts"
+             removeResources(jobExecutionContextsToRemove, resolver)
+             log.info "Removing ${stepExecutionContextsToRemove.size()} StepExecutionContexts"
+             removeResources(stepExecutionContextsToRemove, resolver)
+        }
+    }
+
+    private void removeResources(List<String> resourcePathsToRemove, ResourceResolver resolver) {
+        resourcePathsToRemove.each {
+            Resource resourceToDelete = resolver.getResource(it)
+            if(resourceToDelete) {
+                resolver.delete(resourceToDelete)
+                log.debug "Resource ${it} will be removed"
+            }
+            else {
+                log.info "Resource ${it} doesn't exist"
+            }
+        }
+        resolver.commit()
```

Comment:
@sagarsane does this need a try/catch? If an exception is caught should we `resolver.revert()`?